core/domain/app_feedback_report_validators.py

```diff
@@ -0,0 +1,412 @@
+                ExternalModelReference objects corresponding to the field_name.
+                For examples, all the external ExplorationModels corresponding
+                to a storage model can be associated with the field name
+                'exp_ids'. This dict is used for validation of External Model
+                properties linked to the storage model.
+        """
+        ticket_model_references = (
+            field_name_to_external_model_references['ticket_id'])
+
+        for ticket_model_reference in ticket_model_references:
+            ticket_model = ticket_model_reference.model_instance
+            if ticket_model is None or ticket_model.deleted:
+                model_class = ticket_model_reference.model_class
+                model_id = ticket_model_reference.model_id
+                cls._add_error(
+                    'ticket_id %s' % (
+                        base_model_validators.ERROR_CATEGORY_FIELD_CHECK),
+                    'Entity id %s: based on field ticket_id having'
+                    ' value %s, expected model %s with id %s but it doesn\'t'
+                    ' exist' % (
+                        item.id, model_id, model_class.__name__, model_id))
+
+    @classmethod
+    def _get_external_instance_custom_validation_functions(cls):
+        return [cls._validate_external_ticket_id]
+
+
+class AppFeedbackReportTicketModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating AppFeedbackReportTicketModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        # Valid id:
+        # [ticket_creation_datetime_msec]:[hash(ticket_name)]:[random hash]
+        # We can only validate the timestamp is an int since the id generation
+        # timestamp and the entity creation timestamp differ slightly.
+        regex_string = (
+            '^\\d+\\.[A-Za-z0-9]{1,%s}\\.[A-Za-z0-9]{1,%s}$' % (
+                base_models.ID_LENGTH, base_models.ID_LENGTH))
+        return regex_string
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return [
+            base_model_validators.ExternalModelFetcherDetails(
+                'report_ids',
+                app_feedback_report_models.AppFeedbackReportModel,
+                item.report_ids)]
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [
+            cls._validate_newest_report_timestamp]
+
+    @classmethod
+    def _validate_newest_report_timestamp(cls, item):
+        """Validate that newest_report_timestamp is less than current time and
+        greater than the earliest possible date of submissions (no earlier than
+        March 2021).
+
+        Args:
+            item: datastore_services.Model. AppFeedbackReportTicketModel to
+                validate.
+        """
+        current_datetime = datetime.datetime.utcnow()
+        if item.newest_report_timestamp > current_datetime:
```

Comment:
Similarly, this could be an interval check, instead. However, per my earlier comments we will need to compute 'newest_report_timestamp' based on the report creation date.

Reply:
Hm I think interval checks would be less valid for datetime since the error is a bit different and it might be more useful to understand whether the datetime is relative to the job's run-time or the entity's submission date, WDYT?

Related, we might then remove `newest_report_timestamp` from the model depending on if we want to aggregate stats based on the model's creation date or the Android device's relative submission date (related to [this thread](https://github.com/oppia/oppia/pull/12298/files/999f0ef9cac954d6d80100926db3e29416959056#r607368643)).

Reply:
I think that's reasonable (re: intervals).

Reply:
Ack, will leave this test as is then